src/transformState.js

```diff
@@ -5,7 +5,28 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const prop of actions) {
+    switch (prop.type) {
+      case 'addProperties':
+        for (const extra in prop.extraData) {
+          state[extra] = prop.extraData[extra];
+        }
+        break;
+
+      case 'clear':
+        for (const del in state) {
```

Comment:
we iterate over object keys
```suggestion
        for (const key in state) {
```